src/vhip/mlmodel/gene_features.py

```diff
@@ -131,3 +136,24 @@ def calculate_aa_counts(self) -> None:
             if self.codon_dict[codon] != 0:
                 aa = CODON_TABLE[codon]
                 self.aa_dict[aa] += self.codon_dict[codon]
+
+
+# Define GeneSet class
+class GeneSet:
+    """Class representing a gene set, usually the genes predicted from a genome sequence.
+
+    Args:
+        gene_file (str): Path of annotated genes file containing gene set of interest.
+    """
+
+    def __init__(self, gene_file: str) -> None:
+        """Initialize class variables and read in an annotated genes file, storing Gene objects and metadata in lists."""
+        if not gene_file or not os.path.getsize(gene_file):
+            raise Exception(
+                "Genes file is not provided or empty. Please provide a valid gene file."
+            )
+
+        sequences = read_sequence(gene_file)
+        self.genes: List[Gene] = [Gene(seq) for seq in sequences]
+        self.gene_ids: List[str] = read_headers(gene_file)
+        self.gene_products: List[str] = read_gene_products(gene_file)
```

Comment:
Something I'm wondering: it might be more streamlined to add Gene class attributes for ids and products (default None). That way we can create a list of Gene objects with all of those attributes built in, instead of having separate lists of ids and products as GeneSet attributes. Does that make sense?

Reply:
so you're saying moving those attributes to the Gene class. And then here, if we wanted a list, you could extract them from the list of Gene class?

Reply:
Sounds reasonable. Go for it!